SalesforceCommerceCloud/pwa-kit
# @W-13876268@ Add error handling on PDP

Fixes #1384 

With the removal of getProps in PWA Kit v3 / template-retail-react-app v1, the error handling for when a product or category is not found was lost.

This PR re-adds error handling so that a product or category that doesn't exist returns an HTTP 404 and other errors are also returned.

# Types of Changes




> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# Changes

- Re-adds error handling for product and category on PDP

# How to Test-Drive This PR

- Check out this branch / PR
- Start the app
- Go to http://localhost:3000/global/en-GB/product/does-not-exist
- Go to http://localhost:3000/global/en-GB/category/does-not-exist

Verify you get an HTTP 404 rather than a blank PDP

![Screenshot 2023-08-02 at 5 27 45 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47404250/646b1fbb-9b5b-4263-ad19-f0737c580bd1)
![Screenshot 2023-08-02 at 5 27 22 PM](https://github.com/SalesforceCommerceCloud/pwa-kit/assets/47404250/275ff3d9-ba2a-4695-b31a-4e1535367751)


# Checklists




## General


## Accessibility Compliance

You must check off all items in **one** of the follow two lists:


_or..._


## Localization
